### DISCUSSION TITLE
Frequently Asked Questions

### DISCUSSION THREAD

**Discussion by rusty-snake** [netblue30/firejail#4446] [Wiki]
Discussion for discussions about https://github.com/netblue30/firejail/wiki/Frequently-Asked-Questions.

Predecessor: #2792

**Comment by rusty-snake** [comment]
- [ ] Migrate matching parts to Troubleshooting and use this chance to review.

**Comment by rusty-snake** [comment]
:heavy_check_mark: Done

---

> ## 3D acceleration with nVidia-Hardware
> 
> Users with NVIDIA hardware may need to add `ignore nogroups` to `globals.local`, see [#3144](https://github.com/netblue30/firejail/issues/3144).

@kmk3 Should this be changed to "Before firejail 0.9.68 users with ..."?

edit also

> If you are using NVIDIA proprietary drivers, you must also set ignore noroot in your firefox.local.

**Reply by kmk3** [reply]
@rusty-snake [on Feb 15](https://github.com/netblue30/firejail/discussions/4446#discussioncomment-2182199):

> > ## 3D acceleration with nVidia-Hardware
> > 
> > Users with NVIDIA hardware may need to add `ignore nogroups` to
> > `globals.local`, see [#3144](https://github.com/netblue30/firejail/issues/3144).
> 
> @kmk3 Should this be changed to "Before firejail 0.9.68 users with ..."?

Done (moved to Troubleshooting.md).

> edit also
> 
> > If you are using NVIDIA proprietary drivers, you must also set ignore
> > noroot in your firefox.local.

Done.

**Comment by rusty-snake** [comment]
> ## Cannot open hyperlink with web browser using another application

Update with information from https://github.com/netblue30/firejail/issues/3919#issuecomment-768078226

**Comment by rusty-snake** [comment]
@glitsj16 https://github.com/netblue30/firejail/wiki/Frequently-Asked-Questions#how-do-i-sandbox-applications-started-via-systemd-d-bus-services is a mixup of systemd and dbus. Although systemd depends on dbus and make heavy usage of it, both have distinct configurations.

### D-Bus

`/usr/share/dbus-1/services/*.service` and `~/.local/share/dbus-1/services/*.service` are used by dbus (dbus-deamon or dbus-broker, whichever implementation is installed by your distribution). To "fix" them, you copy from the system directory to the user directory, adjust the `Exec` line and restart your dbus-deamon/dbus-broker (e.g. `systemctl --user restart dbus.service`, the `dbus.service` alias should work for both implementations; IDK if a `reload` works too). *This can all be done in the absence of systemd* like on runit systems.

### systemd --user

`/usr/lib/systemd/user/*.service` and `~/.config/systemd/user/*.service` are used by systemd --user. To "fix" them, you copy from the system directory to the user directory, adjust the `Exec` line and `systemctl --user daemon-reload`. The path in `/usr/share/dbus-1/services/foobar.service` does not matter unless it is a `Type=DBus` service.

**Reply by ghost** [reply]
Very clear on the distictions between the two!
On a side-note, I'm having some trouble creating a working uri-fragment on the wiki to the entree in the FAQ. In HTML it's easy: `<a name="how-do-i-sandbox-systemd-started-applications"></a>`. Would be very helpful if you could show how to do that exactly in the GH markdown editor.

**Reply by rusty-snake** [reply]
![screenshot](https://github.com/netblue30/firejail/assets/41237666/a184b7db-e1bf-40b6-9f72-97a6ec6b13f2)

**Reply by ghost** [reply]
Reworked the FAQ entree, including the now working uri-fragment. Thanks again for sharing your excellent insights!

**Reply by curiosityseeker** [reply]
Thanks for this! However, I'm pretty sure that most users (myself included) do not know what this "fix" actually means. What exactly is "fixed" and to what purpose? I suggest to clarify this in the wiki.

**Reply by ghost** [reply]
@curiosityseeker

By default these kind of service files start applications `non-sandboxed`, due to using full paths that bypass those created via firecfg (or otherwise). The purpose of creating local copy's and editing those is to `force` the applications in question running via `firejail`. In the 'heat' of the moment I forgot to add this. Will "fix" the "fixes" :D shortly.

**Reply by ghost** [reply]
@curiosityseeker

I've added some context. HTH. Please report back if it's still unclear.

**Reply by curiosityseeker** [reply]
@glitsj16 
Thanks a lot for these additional remarks which clarify things!

Note, though (and just nickpitting), that at the bottom it says "Currently the firecfg command doesn't do any of this automatically. For an alternative implementation that supports these features, see [firecfg.py](https://github.com/rusty-snake/firecfg.py)." 

However, the firecfg.py site [says](https://github.com/rusty-snake/firecfg.py#features): 

> TODO: Firejail systemd --user
TODO: Firejail dbus-system-service

So it seems that `firecfg.py` is not yet a fully comprehensive alternative. But I'm absolutely confident that @rusty-snake will add this before long. :+1:
